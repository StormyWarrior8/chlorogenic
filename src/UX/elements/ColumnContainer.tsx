import CloseIcon from "mdi-react/CloseIcon"
import React from "react"
import { Flexbox } from ".."
import styled from "../Styled"
import Button from "./Button"
import { VFlexbox } from "./Flexbox"

const ColumnContainer = styled(VFlexbox)`
  border: solid 1px ${({ theme }) => theme.baseColor};
  border-radius: 4px;
  padding: 0.4em;
  margin: 2px;
  background: ${({ theme }) => theme.backgroundColor};
  width: 16em;
`

ColumnContainer.displayName = "ColumnContainer"

const Scroller = styled(Flexbox)`
  height: 100%;
  overflow-x: auto;
  flex-direction: column;
`

const Header = styled(Flexbox)`
  margin: 0;
  font-size: large;
  margin-bottom: 1em;
  align-items: flex-start;
  h3 {
    margin: 0;
  }

  button {
    margin: 0;
    padding: 0;
    margin-left: auto;
    &:hover {
      background: ${({ theme }) => theme.redColor};
    }
  }
`

const Description = styled.p`
  margin: 0;
  color: ${({ theme }) => theme.textColor};
  font-size: small;
  text-align: right;
`

export interface ColumnContainerProps {
  header: string
  description?: string
  onClickClose?(): void
}

const Layouted: React.SFC<ColumnContainerProps> = ({
  header,
  description,
  onClickClose,
  children,
}) => (
  <ColumnContainer>
    <Header>
      <h3>{header}</h3>
      {onClickClose && (
        <Button size="small" transparent={true} onClick={onClickClose}>
          <CloseIcon size={12} />
        </Button>
      )}
    </Header>
    {description && <Description>{description}</Description>}
    <Scroller>{children}</Scroller>
  </ColumnContainer>
)
export default Layouted
