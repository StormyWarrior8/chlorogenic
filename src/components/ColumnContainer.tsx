import * as React from "react"
import styled from "../appearance/styled"
import { Icon } from "./parts"
import Button from "./parts/Button"
import Flexbox, { VFlexbox } from "./parts/Flexbox"

const ColumnContainer = styled(VFlexbox)`
  border: solid 1px ${({ theme }) => theme.baseColor};
  border-top: 0;
  border-bottom: 0;
  padding: 0.6em;
  padding-bottom: 0;
  background: ${({ theme }) => theme.backgroundColor};
  width: 16em;
  flex-shrink: 0;
  margin-left: 0.2em;
`

ColumnContainer.displayName = "ColumnContainer"

const Scroller = styled(Flexbox)`
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  flex-direction: column;
`

const Header = styled(Flexbox)`
  margin: 0;
  font-size: large;
  align-items: flex-start;
  h3 {
    margin: 0;
  }

  button {
    margin: 0;
    padding: 0;
    margin-left: auto;
    &:hover {
      background: ${({ theme }) => theme.redColor};
    }
  }
`

const Description = styled.p`
  white-space: pre;
  margin: 0;
  color: ${({ theme }) => theme.secondaryTextColor};
  font-size: small;
  text-align: right;
`

export interface ColumnContainerProps {
  header: string
  description?: string
  onClickClose?(): void
}

const Layouted: React.SFC<ColumnContainerProps> = ({
  header,
  description,
  onClickClose,
  children,
}) => (
  <ColumnContainer>
    <Header>
      <h3>{header}</h3>
      {onClickClose && (
        <Button size="small" transparent={true} onClick={onClickClose}>
          <Icon type="close" size={0.6} />
        </Button>
      )}
    </Header>
    {description && <Description>{description}</Description>}
    <Scroller>{children}</Scroller>
  </ColumnContainer>
)
export default Layouted
