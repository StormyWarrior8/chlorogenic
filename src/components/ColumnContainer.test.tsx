import * as React from "react"
import { render } from "react-testing-library"
import ColumnContainer from "./ColumnContainer"
it("snapshot", () => {
  {
    const { container } = render(<ColumnContainer header="header text" />)
    expect(container.firstChild).toMatchSnapshot()
  }
})
