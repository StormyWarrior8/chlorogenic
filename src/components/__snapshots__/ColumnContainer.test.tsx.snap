// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="sc-ifAKCX hyrIVx sc-bxivhb glCqXO"
>
  <div
    class="sc-bZQynM iEfYBC sc-htpNat cixthU"
  >
    <h3>
      header text
    </h3>
  </div>
  <div
    class="sc-EHOje gGXDvz sc-bxivhb glCqXO"
  />
</div>
`;
